File: Python/program/Prodi.py
# class declaration
class Prodi:
    # atribut private
    __nama_prodi = ""
    __kode = ""
    # composite kelas course
    __Course = []  # array of object course

   # konstruktor
    def __init__(self, nama_prodi="", kode="",  Course=[]):
        self.__nama_prodi = nama_prodi
        self.__kode = kode
        self.__Course = Course

    # Getter and Setter
    def get_nama_prodi(self):
        return self.__nama_prodi

    def set_nama_prodi(self, nama_prodi):
        self.__nama_prodi = nama_prodi

    def get_kode(self):
        return self.__kode

    def set_kode(self, kode):
        self.__kode = kode

    def get_Course(self):
        return self.__Course

    def set_Course(self, Course):
        self.__Course = Course

File: CPP/program/Prodi.cpp
// import file header
#include "header.hh"

/* konstruktor */

// konstruktor tanpa parameter
Prodi::Prodi()
{
    // set isi atribut secara default

    this->nama_prodi = "";
    this->kode = "";
}

// konstruktor dengan parameter
Prodi::Prodi(string nama_prodi, string kode, Course matkul)
{
    // set isi atribut berdasarkan isi parameter
    this->nama_prodi = nama_prodi;
    this->kode = kode;
    this->matkul.push_back(matkul);
}

/* Getter dan Setter*/

// Get nama_prodi
string Prodi::get_nama_prodi()
{
    return this->nama_prodi;
}

// Set nama_prodi
void Prodi::set_nama_prodi(string nama_prodi)
{
    this->nama_prodi = nama_prodi;
}
// Get kode
string Prodi::get_kode()
{
    return this->kode;
}

// Set kode
void Prodi::set_kode(string kode)
{
    this->kode = kode;
}

// Get MataKuliah
list<Course> Prodi::get_matkul()
{
    return this->matkul;
}

// Set MataKuliah
void Prodi::set_matkul(list<Course> matkul)
{
    this->matkul = matkul;
}
// add list objek MataKuliah
void Prodi::add_matkul(Course matkul)
{
    this->matkul.push_back(matkul);
}

/* desktruktor */

// default
Prodi::~Prodi() // desktruktor
{
}
